### PR TITLE
🐛 Retrieving service key/version from running service to use regex instead of split

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- an issue arose where an image name used in a dynamic service was not following the conventions but of the type ```simcore/services/dynamic/jupyter-smash:2.5.8@sha256:lksdfjhlskfjsdlfksdjflgsdkjhgfsd123eu210```, this triggered a weird situation where the director-v0 was not able to stop the service, and letting a study as un-closed.

This PR fixes this problem.

### CODE REVIEW

File: services/director/src/simcore_service_director/producer.py [hunk 1]
```diff
@@ -817,7 +822,15 @@ async def _get_service_key_version_from_docker_service(
...
     service_full_name = service_full_name[len(config.REGISTRY_PATH) :].strip("/")
-    return service_full_name.split(":")[0], service_full_name.split(":")[1]
+    service_re_match = _SERVICE_KEY_REGEX.match(service_full_name)
+    if not service_re_match:
+        raise exceptions.DirectorException(
+            msg=f"service {service_full_name} does not contain valid simcore key"
+        )
+    service_key = service_re_match.group(1)
+    service_tag = service_re_match.group(4)
+    # FIXME: service_key:service_tag@sha256:230498230492834
```
Comment (pcrespov) [inline services/director/src/simcore_service_director/producer.py:832]: - remove FIXME comment
- add what a "valid simcore key" means in the exception message e.g. 
```
f"'{service_full_name}' in an invalid service name, expected name as 'simcore/services/comp/myservice:1.2.3' "
```

**Comment by mguidon** [comment]
So you are saying the service tag of the jupyter-smash also contained the sha? This should never have happened.

**Comment by pcrespov** [comment]
> So you are saying the service tag of the jupyter-smash also contained the sha? This should never have happened.

@mguidon we need the audit tooling of the registry and a stronger validation-for-push in place ASAP. This bug was "funny" to find :-)

**Comment by mguidon** [comment]
> > So you are saying the service tag of the jupyter-smash also contained the sha? This should never have happened.
> 
> @mguidon we need the audit tooling of the registry and a stronger validation-for-push in place ASAP. This bug was "funny" to find :-)

Do we know how that happened? And is it the only service?

**Review by pcrespov — APPROVED** [review]
🎉

**Comment by pcrespov** [comment]
> Do we know how that happened? And is it the only service?

No. 
I do not not know. We will have to check tomorrow.

**Review by GitHK — APPROVED** [review]
👍

**Pull request merged**